packages/esm-patient-tests-app/src/test-results/filter/filter-context.tsx

```diff
@@ -126,17 +128,24 @@ const FilterProvider = ({ roots, isLoading, children }: FilterProviderProps) =>
 
   const tableData = useMemo<GroupedObservation[]>(() => {
     const flattenedObs: Observation[] = [];
+    const seenTests = new Set<string>();
 
     for (const key in state.tests) {
       const test = state.tests[key] as TestResult;
       if (test.obs && Array.isArray(test.obs)) {
         test.obs.forEach((obs) => {
-          const flattenedEntry = {
-            ...obs,
-            key: key,
-            ...test,
-          };
-          flattenedObs.push(flattenedEntry);
+          // Use a more specific key that includes the test name to avoid over-deduplication
+          const testKey = `${test.flatName}_${obs.obsDatetime}_${obs.value}`;
+
+          if (!seenTests.has(testKey)) {
```

Comment:
The other option of preventing duplicates is changing the data structure of `flattenedObs` to a Set.

Reply:
Wouldn't the set require the same deduplication logic? Plus, the set only checks object references to determine uniqueness, whereas here we'd need to deduplicate based on content instead.